### DISCUSSION TITLE
Testing set functions

### DISCUSSION THREAD

**Discussion by dkgrieshammer** [pmndrs/zustand#247] [Q&A]
Hi Folks, any hints regarding testing set functions?
How would you go about the example functions from the repo since they obviously have side effects? i.e.:

increasePopulation: () => set(state => ({ bears: state.bears + 1 }))

Would be really grateful for any hints, stackoverflow passed on me this time :)

**Comment by dai-shi** [comment]
I'm not very experienced in testing with zustand, but if I were to test functions in the store, I would define them outside and test separately.

```js
const actions = (set) => ({
  increasePopulation: () => set(state => ({ bears: state.bears + 1 }))
})

const useStore = create((set) => ({
  bears: 0,
  ...actions(set),
}))
```

Or, I'd test react component with useStore.

**Reply by dkgrieshammer** [reply]
Thanks for the answer, still the state is a side-effect; I think I got a way by using reducers; they can be tested separately; avoided them until this point but they might be the answer to clean jest testing. I'll try, thx 🙏
